crates/scroll/node/src/builder/network.rs

```diff
@@ -79,33 +135,192 @@ where
+                    }) {
+                        Ok(sig) => sig,
+                        Err(e) => {
+                            warn!(
+                                "Failed to get block signature from database, header hash: {:?}, error: {}",
+                                header.hash_slow(),
+                                HeaderTransformError::DatabaseError(e.to_string())
+                            );
+                            None
+                        }
+                    }
+                } else {
+                    warn!(
+                        "Failed to get block signature from database, header hash: {:?}, error: {}",
+                        header.hash_slow(),
+                        HeaderTransformError::NoRuntimeAvailable
+                    );
+                    None
+                }
+            });
+            if let Some(sig) = signature {
+                let authorized_signer = match self.chain_spec.chain().named() {
+                    Some(NamedChain::Scroll) => Some(SCROLL_MAINNET_SIGNER),
+                    Some(NamedChain::ScrollSepolia) => Some(SCROLL_SEPOLIA_SIGNER),
+                    _ => None,
+                };
+
+                if let Err(err) = recover_and_verify_signer(&sig, header.hash_slow(), authorized_signer) {
+                    warn!(
+                        "Found invalid signature(different from the hardcoded signer) for header hash: {:?}, sig: {:?}, error: {}",
+                        header.hash_slow(),
+                        sig.to_string(),
+                        err
+                    );
+                    return H::default();
+                }
+
+                *header.extra_data_mut() = sig.as_bytes().into();
+            }
         }
         header
     }
 }
+
+/// Recover signer from signature and verify authorization.
+fn recover_and_verify_signer(
+    signature: &Signature,
+    hash: B256,
+    authorized_signer: Option<Address>,
+) -> Result<Address, HeaderTransformError> {
+    // Recover signer from signature
+    let signer = reth_primitives_traits::crypto::secp256k1::recover_signer(
+        signature,
+        hash,
+    )
+    .map_err(|_| HeaderTransformError::RecoveryFailed)?;
+
+    // Verify signer is authorized
+    if authorized_signer.is_some() && authorized_signer.unwrap() != signer {
+        return Err(HeaderTransformError::InvalidSigner(signer));
+    }
+
+    Ok(signer)
+}
+
+/// Parse a canonical 65-byte secp256k1 signature: r (32) | s (32) | v (1).
+fn parse_65b_signature(bytes: &[u8]) -> Result<Signature, HeaderTransformError> {
+    if bytes.len() != 65 {
+        return Err(HeaderTransformError::InvalidSignature);
+    }
+
+    let signature = Signature::from_raw(&bytes)
+        .map_err(|_| HeaderTransformError::InvalidSignature)?;
+
+    Ok(signature)
+}
+
+/// Run the async DB insert from sync code safely.
+fn persist_signature_blocking(
```

Comment:
This is not blocking anymore with the migration to `tokio::spawn` so lets update the function name to reflect that.
```suggestion
fn persist_signature(
```